From 4d489b635d6b3a1f93814ac2a34c88a78fd63780 Mon Sep 17 00:00:00 2001
From: zigen <hrlclb@gmail.com>
Date: Wed, 17 Aug 2022 13:24:14 +0900
Subject: [PATCH 6/8] fix msgbox mainloop error

---
 src/qtenv/qtenv.cc | 22 +++++++++++++++++++---
 1 file changed, 19 insertions(+), 3 deletions(-)

diff --git a/src/qtenv/qtenv.cc b/src/qtenv/qtenv.cc
index 06191ea372..9c7fad7f1e 100644
--- a/src/qtenv/qtenv.cc
+++ b/src/qtenv/qtenv.cc
@@ -2406,11 +2406,27 @@ void Qtenv::confirm(DialogKind kind, const char *msg)
         out << "\n<!> " << prefix << msg << endl << endl;
     }
     else {
+        QMessageBox* msgBox = nullptr;
         switch (kind) {
-        case INFO: QMessageBox::information(mainWindow, "Confirm", msg, QMessageBox::StandardButton::Ok); break;
-        case WARNING: QMessageBox::warning(mainWindow, "Confirm", msg, QMessageBox::StandardButton::Ok); break;
-        case ERROR: QMessageBox::critical(mainWindow, "Confirm", msg, QMessageBox::StandardButton::Ok); break;
+        case INFO:
+            msgBox = new QMessageBox(QMessageBox::Icon::Information, "Confirm", msg, QMessageBox::StandardButton::Ok, mainWindow);
+            break;
+        case WARNING:
+            msgBox = new QMessageBox(QMessageBox::Icon::Warning, "Confirm", msg, QMessageBox::StandardButton::Ok, mainWindow);
+            break;
+        case ERROR:
+            msgBox = new QMessageBox(QMessageBox::Icon::Critical, "Confirm", msg, QMessageBox::StandardButton::Ok, mainWindow);
+            break;
+        }
+        if (msgBox == nullptr) return;
+        bool accepted = false;
+        connect(msgBox, &QMessageBox::accepted, [&]() {accepted = true;});
+        msgBox->open();
+        msgBox->adjustSize();
+        while(!accepted) {
+            emscripten_sleep(10);
         }
+        delete msgBox;
     }
 }
 
-- 
2.39.0

