diff --git a/configure b/configure
index 7db1e1c187..d5ee1420a0 100755
--- a/configure
+++ b/configure
@@ -3151,10 +3151,10 @@ $as_echo "no" >&6; }
 fi
 
 
-if test "$tmp_CXX" = "not found"; then
-   OPP_PROBLEM_PACKAGES="$OPP_PROBLEM_PACKAGES g++"
-   as_fn_error $? "No C++ compiler found -- one is required to build OMNeT++/OMNEST, and also for model development." "$LINENO" 5
-fi
+#if test "$tmp_CXX" = "not found"; then
+#   OPP_PROBLEM_PACKAGES="$OPP_PROBLEM_PACKAGES g++"
+#   as_fn_error $? "No C++ compiler found -- one is required to build OMNeT++/OMNEST, and also for model development." "$LINENO" 5
+#fi
 
 
 #------------------------------------
